src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -157,5 +160,406 @@ ConvSolution InitAnyInvokerFactory(const ProblemDescriptionType& problem,
     return result;
 }
 
+struct TransposeInstance
+{
+    size_t tensor_sz = 0;
+    std::vector<OpKernelArg> kern_args{};
+    int index         = -1;
```

Comment:
This one is important.
```suggestion
    size_t index         = std::numeric_limits<size_t>::max();
```
It has `unsigned` nature and will be properly compared with `unsigned` `kernels.size()`.

Reply:
using an `int` here simplifies things. `-1` is a sentinel value. The check later on against `kernel.size()` serves another purpose as well, which is that the vector being passed is of the right size.

Reply:
@amberhassaan I am second to @CAHEK7. Let's avoid mixing signed and unsigned types wherever possible -- _that_ simplifies things IMHO. Otherwise the future code readers must always have in mind types of operands and implicit conversion rules (instead of focusing on more important things).

Reply:
fixed in latest code.